src/cargo/ops/fix.rs

```diff
@@ -738,7 +738,18 @@ fn rustfix_and_fix(
         // As mentioned above in `rustfix_crate`, we don't immediately warn
         // about suggestions that fail to apply here, and instead we save them
         // off for later processing.
+        let mut already_applied = HashSet::new();
         for suggestion in suggestions.iter().rev() {
+            // Skip seemly duplicated suggestions.
+            if suggestion
+                .solutions
+                .iter()
+                .any(|sol| !already_applied.insert(sol))
+            {
+                // already applied
+                fixed_file.fixes_applied += 1;
```

Comment:
Why are we incrementing this if nothing happened? It seems weird in the test to say two fixes were applied when really only one change was made.

Reply:
I was wondering if we match the number of warnings Cargo show, but after seeing the actual message Cargo emitted, it seems better not to increment.

```
warning: `foo` (lib) generated 2 warnings (run `cargo fix --lib -p foo` to apply 2 suggestions)
```

Reply:
Actually, I am not so sure…

Reply:
Anyway, I think it should reflect the truth at this moment. Updated.